include/circt/Dialect/Pipeline/PipelineOps.td

```diff
@@ -225,9 +240,19 @@ def StageOp : Op<Pipeline_Dialect, "stage", [
     void setNextStage(Block *block) {
       setSuccessor(block);
     }
+
+    // Returns the list of clock gates for the given register.
+    ValueRange getClockGatesForReg(unsigned regIdx);
   }];
-}
 
+  let builders = [
+    OpBuilder<(ins "Block*":$dest, "ValueRange":$registers, "ValueRange":$passthroughs)>,
+    // Clock gate builder, which takes a mapping between the registers and
+    // and their clock gate hierarchy.
+    OpBuilder<(ins "Block*":$dest, "ValueRange":$registers, "ValueRange":$passthroughs,
```

Comment:
Does it make sense to add a builder which associates the registers with the clock gates with a struct? (i.e. `struct {Value reg; vector clockGates;}`.)

Reply:
I could... but the highlighted builder isn't even being used 👀 - not a big fan of adding too much code which won't be tested.